hpedockerplugin/file_manager.py

```diff
@@ -0,0 +1,479 @@
+import base64
+import copy
+import json
+import socket
+import string
+import sh
+import six
+from Crypto.Cipher import AES
+from threading import Thread
+
+from oslo_log import log as logging
+from oslo_utils import netutils
+
+from hpedockerplugin.cmd import cmd_createshare
+from hpedockerplugin.cmd import cmd_generate_fpg_vfs_names
+from hpedockerplugin.cmd import cmd_setquota
+from hpedockerplugin.cmd import cmd_deleteshare
+
+import hpedockerplugin.exception as exception
+import hpedockerplugin.fileutil as fileutil
+import hpedockerplugin.hpe.array_connection_params as acp
+from hpedockerplugin.i18n import _
+from hpedockerplugin.hpe import hpe_3par_mediator
+from hpedockerplugin import synchronization
+
+LOG = logging.getLogger(__name__)
+
+
+class FileManager(object):
+    def __init__(self, host_config, hpepluginconfig, etcd_util,
+                 fp_etcd_client, node_id, backend_name='DEFAULT'):
+        self._host_config = host_config
+        self._hpepluginconfig = hpepluginconfig
+        self._my_ip = netutils.get_my_ipv4()
+
+        self._etcd = etcd_util
+        self._fp_etcd_client = fp_etcd_client
+        self._node_id = node_id
+        self._backend = backend_name
+
+        self._initialize_configuration()
+
+        self._decrypt_password(self.src_bkend_config,
+                               self.tgt_bkend_config,
```

Comment:
@ For the current implementation this is unused parameter, can we remove it?

Reply:
Replication was one of the initial asks. Hence the code to support that was left as part of the implementation. We can remove it now as the requirement doesn't hold good anymore.